crates/orchestrator/src/workers/mod.rs

```diff
@@ -1,13 +1,39 @@
-use std::error::Error;
-
+use crate::{config::config, jobs::types::JobStatus};
 use async_trait::async_trait;
+use std::error::Error;
 
+pub mod data_submission;
 pub mod proof_registration;
 pub mod proving;
 pub mod snos;
 pub mod update_state;
 
 #[async_trait]
 pub trait Worker: Send + Sync {
+    async fn run_worker_if_enabled(&self) -> Result<(), Box<dyn Error>> {
+        if !self.is_worker_enabled().await? {
+            return Ok(());
+        }
+        self.run_worker().await
+    }
+
     async fn run_worker(&self) -> Result<(), Box<dyn Error>>;
+
+    // Assumption : False Negative
+    // we are assuming that the worker will spawn only 1 job for a block and no two jobs will ever exist
```

Comment:
im not 100% sure of this assumption, like jobs in my mind should be immutable. It is quite error prone to do the opposite.

Reply:
for me once a job has failed, it's a failed state and the only way ahead is to spawn a new one, wdyt @apoorvsadana

Reply:
So creating a new job as a whole might be difficult because we don't know at what stage the job failed. Currently, I was thinking we go ahead with this -> investigate what sort of failures we're getting and add new code accordingly to retry failed jobs (so you don't need to manually edit the DB but we add code on demand to handle failed cases based on what we see)

Reply:
yes I think that's already a better alternative, would be great to have clear error codes listed